### PR TITLE
Automated chained lifecycle environments table fits screen

### LINKED ISSUE #3215: Need UI test for Lifecycle environments visibility if there are more environments
See BZ #1295922 for more information.  If an environment path has more environments, eventually it runs out of space in UI and there is no way for user to scroll to the right since the scrollbar is not shown currently.

### PR DESCRIPTION
**Pull request by renzon**
close #3215

Test result:

```console
/home/renzo/PycharmProjects/venvs/robottelo/bin/python /home/renzo/bin/pycharm-2016.3/helpers/pycharm/pytestrunner.py -p pytest_teamcity /home/renzo/PycharmProjects/robottelo/tests/foreman/ui/test_lifecycleenvironment.py "-k LifeCycleEnvironmentTestCase and test_positive_env_list_fits_screen"
============================== 5 tests deselected ==============================
========= 1 passed, 5 deselected, 1 pytest-warnings in 241.74 seconds ==========
```

### CODE REVIEW

File: tests/foreman/ui/test_lifecycleenvironment.py [hunk 1]
```diff
@@ -15,13 +15,18 @@
 
 @Upstream: No
 """
+from itertools import chain
 
-from fauxfactory import gen_string
+from fauxfactory import gen_alpha, gen_string
```
Comment (oshtaier) [inline tests/foreman/ui/test_lifecycleenvironment.py]: don't need to import `gen_alpha` use `gen_string('alpha')` instead

File: tests/foreman/ui/test_lifecycleenvironment.py [hunk 2]
```diff
@@ -15,13 +15,18 @@
...
 from nailgun import entities
+
 from robottelo.datafactory import generate_strings_list
-from robottelo.decorators import run_only_on, stubbed, tier1, tier2
+from robottelo.decorators import run_only_on, stubbed, tier1, tier2, \
```
Comment (oshtaier) [inline tests/foreman/ui/test_lifecycleenvironment.py]:
```
from robottelo.decorators import (
    run_only_on, 
    skip_if_bug_open,
    stubbed, 
    tier1, 
    tier2,
)   
```

File: tests/foreman/ui/test_lifecycleenvironment.py [hunk 3]
```diff
@@ -15,13 +15,18 @@
...
+    skip_if_bug_open
 from robottelo.test import UITestCase
 from robottelo.ui.factory import make_lifecycle_environment
+from robottelo.ui.locators.base import locators
```
Comment (oshtaier) [inline tests/foreman/ui/test_lifecycleenvironment.py]: `from robottelo.ui.locators import common_locators, locators`

File: tests/foreman/ui/test_lifecycleenvironment.py [hunk 4]
```diff
@@ -154,3 +159,34 @@ def test_positive_create_environment_after_host_register(self):
...
+        """Check if long list of lifecycle environments fits into screen
+
+        @id: 63b985b0-c847-11e6-92ad-68f72889dc7f
+
+        @Setup: save 8+ chained lifecycles environments
+
+        @BZ: 1295922
+
+        @Assert: lifecycle enviroments table fits screen
+
```
Comment (oshtaier) [inline tests/foreman/ui/test_lifecycleenvironment.py]: redundant line

File: tests/foreman/ui/test_lifecycleenvironment.py [hunk 5]
```diff
@@ -154,3 +159,34 @@ def test_positive_create_environment_after_host_register(self):
 
         @caseautomation: notautomated
         """
+
+    @tier1
+    @skip_if_bug_open('bugzilla', 1295922)
```
Comment (oshtaier) [inline tests/foreman/ui/test_lifecycleenvironment.py]: Closed a long time ago, so don't need to have BZ decorator

File: tests/foreman/ui/test_lifecycleenvironment.py [hunk 6]
```diff
@@ -154,3 +159,34 @@ def test_positive_create_environment_after_host_register(self):
...
+                    name=name,
+                    prior=prior
+                )
+            envs_table = session.nav.wait_until_element(
+                locator=locators['content_env.table']
+            )
+            table_width = envs_table.size['width']
+            body = session.nav.find_element(common_locators['body'])
+            body_width = body.size['width']
+            self.assertGreaterEqual(body_width - table_width, 0)
```
Comment (oshtaier) [inline tests/foreman/ui/test_lifecycleenvironment.py]: Nice idea. Could you please provide example of numbers (like `1920-1500`?). Not sure on 100% whether visible area and width is the same, but at least one aspect can be verified

Comment (oshtaier) [inline tests/foreman/ui/test_lifecycleenvironment.py]: Especially example when we have let's say 5 and 10 envs.

**Comment by oshtaier** [comment]
And maybe not `fits screen`, but `fits browser screen` or something like this as probably these are completely different things.

File: tests/foreman/ui/test_lifecycleenvironment.py (hunk 6 above)
Comment (renzon) [inline tests/foreman/ui/test_lifecycleenvironment.py]: The problem is that they fixed this issue changing css of table to 100% (widh and max-width). So I would need a 6.1 box to test it. Going to check if anyone has  a box like that.

**Comment by renzon** [comment]
@oshtaier fixed your comments, dealing only with number examples.

File: tests/foreman/ui/test_lifecycleenvironment.py (hunk 6 above)
Comment (oshtaier) [inline tests/foreman/ui/test_lifecycleenvironment.py]: Ahh, I see. At least, provide an example for working scenario

Comment (renzon) [inline tests/foreman/ui/test_lifecycleenvironment.py]: on sat 6.3 the widths are constants (on my maximized screen): table: 1334 body:1366

Comment (renzon) [inline tests/foreman/ui/test_lifecycleenvironment.py]: Ok, it worked on 6.1: 
table width: 2151
body width: 1600

Failing Results:

```console
/home/renzo/PycharmProjects/venvs/robottelo/bin/python /home/renzo/bin/pycharm-2016.3/helpers/pycharm/pytestrunner.py -p pytest_teamcity /home/renzo/PycharmProjects/robottelo/tests/foreman/ui/test_lifecycleenvironment.py "-k LifeCycleEnvironmentTestCase and test_positive_env_list_fits_browser_screen"

>           self.assertGreaterEqual(body_width - table_width, 0)
E           AssertionError: -551 not greater than or equal to 0
========= 1 failed, 5 deselected, 1 pytest-warnings in 148.16 seconds ==========
```

Comment (renzon) [inline tests/foreman/ui/test_lifecycleenvironment.py]: For sat 6.3 tests passed (they are on PR description) and number are:
table width: 1334
body width: 1366

**Comment by oshtaier** [comment]
Thanks @renzon for details!

**Comment by renzon** [comment]
@oshtaier I have made the changes you asked for, can you review this?

**Comment by oshtaier** [comment]
@renzon ?. It is 10 days passed since PR was merged, so not sure what I should do

**Pull request merged**